R/ExpHazOS.R

```diff
@@ -0,0 +1,17 @@
+#' OS Hazard Function from Constant Transition Hazards
+#'
+#' @param t (`numeric`)\cr study time-points.
+#' @param h01 (positive `number`)\cr transition hazard for 0 to 1 transition.
+#' @param h02 (positive `number`)\cr transition hazard for 0 to 2 transition.
+#' @param h12 (positive `number`)\cr transition hazard for 1 to 2 transition.
+#'
+#' @return This returns the value of the OS hazard function at time t.
+#' @export
+#'
+#' @examples
+#' ExpHazOS(c(1:5), 0.2, 1.1, 0.8)
+ExpHazOS <- function(t, h01, h02, h12) {
+  h012 <- h12 - h01 - h02
+  res <- ((h12 - h02) * (h01 + h02) - h01 * h12 * exp(-h012 * t)) / ((h12 - h02) - h01 * exp(-h012 * t))
+  return(res)
```

Comment:
avoid explicit return
```suggestion
  ((h12 - h02) * (h01 + h02) - h01 * h12 * exp(-h012 * t)) / ((h12 - h02) - h01 * exp(-h012 * t))
```